### PR TITLE
feat: chart analytics

### LINKED ISSUE #452: ⚡ Feature | Chart design for Analytics page
Step 3 for the Analytics page is to display the graph with 2 data, total visitors and total views.

- The total number of visitors is the total number of unique users who visited a given profile.

- The total number of views corresponds to all the user clicks on a given profile (each time a user clicks on a published story for example).

On this issue, we are focusing on the chart itself and **not on the number display**:
Capture d’écran 2022-05-02 à 12 41 52

**Use that CodeSandBox to make the chart:**
https://codesandbox.io/s/hrje5

Link to Figma page: https://www.figma.com/file/elxKahbV8rj85IWOEyZeVu/Product-design?node-id=1787%3A12400

Estimate: 3D

### PR DESCRIPTION
**Pull request by gregogun**
Adds tabbable graph views to display the views and visits stats for the user's blog. 

Closes #452

### CODE REVIEW

File: sigle/src/modules/analytics/stats/StatsFrame.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    });
+
+    setData(stats);
+  };
+
+  const fetchMonthlyStats = async () => {
+    setData(initialRange);
+
+    const statsRes = await fetch(
+      `${baseUrl}/api/analytics/historical?dateFrom=${format(
```
Comment (pradel) [inline sigle/src/modules/analytics/stats/StatsFrame.tsx:69]: I think this should work fine too :) 
```suggestion
      `/api/analytics/historical?dateFrom=${format(
```

File: sigle/src/modules/analytics/stats/utils.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { defaultStyles } from '@visx/tooltip';
```
Comment (pradel) [inline sigle/src/modules/analytics/stats/utils.ts:1]: as the utils are just for the tooltip should we rename it to tooltip.ts? or tooltipUtils?

File: sigle/src/modules/analytics/stats/StatsFrame.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+          <TabsContent value="weekly">
+            <Box
+              css={{
+                mb: '$5',
+                position: 'relative',
+                width: '100%',
+                height: 400,
+              }}
+            >
+              <StatsChart type="week" data={data} />
```
Comment (pradel) [inline sigle/src/modules/analytics/stats/StatsFrame.tsx:147]: As we are using tabs, the StatsChart element is recreated and destroyed every time the type is changing. I am wondering if we were rendering the same view but just changing the props passed down to StatsChart, would the graph change the values with a nice animation?

File: sigle/src/modules/analytics/stats/StatsFrame.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        fetchWeeklyStats();
+        break;
+      case 'monthly':
+        fetchMonthlyStats();
+        break;
+      case 'all':
+        fetchAllTimeStats();
+        break;
+      default:
+        fetchWeeklyStats();
```
Comment (pradel) [inline sigle/src/modules/analytics/stats/StatsFrame.tsx:117]: Default should probably throw an error as this should not happen :)

File: sigle/src/modules/analytics/stats/StatsFrame.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+export const StatsFrame = () => {
+  const [data, setData] = useState<StatsData[]>(initialRange);
+
+  const baseUrl = window.location.origin;
+
+  useEffect(() => {
+    fetchWeeklyStats();
+  }, []);
+
+  const fetchWeeklyStats = async () => {
```
Comment (pradel) [inline sigle/src/modules/analytics/stats/StatsFrame.tsx:43]: by looking at the code it feels like fetchWeeklyStats, fetchMonthlyStats and fetchAllTimeStats could be merged into one function as they are doing the same thing, wdyt?

File: sigle/src/types/index.ts [hunk 6]
```diff
@@ -103,3 +103,26 @@ export interface SettingsFile {
    */
   siteTwitterHandle?: string;
 }
+
+export interface AnalyticsHistoricalData {
```
Comment (pradel) [inline sigle/src/types/index.ts:107]: As these types are only used by the analytics, the types should probably stay within the stats module

File: sigle/src/types/index.ts [hunk 7]
```diff
@@ -103,3 +103,26 @@ export interface SettingsFile {
...
+  pageviews: number;
+}
+
+export type AnalyticsHistoricalResponse = {
+  historical: AnalyticsHistoricalData[];
+  stories: AnalyticsStoriesData[];
+};
+
+export interface StatsData {
+  value: number;
```
Comment (pradel) [inline sigle/src/types/index.ts:125]: to avoid confusion, should this be renamed to pageviews?

**Review by pradel — CHANGES_REQUESTED** [review]
Good job this one was a big one 🚀 

I found the following bug while testing, my mouse is on the line, showing the 83 views tooltip, but the 83 pick is actually on the right side. So the tooltip is not showing exactly what you are hovering.

Screenshot 2022-05-20 at 10 12 16

File: sigle/src/modules/analytics/stats/StatsFrame.tsx (hunk 5 above)
Comment (gregogun) [reply sigle/src/modules/analytics/stats/StatsFrame.tsx:43]: agreed, will merge them :)

File: sigle/src/types/index.ts (hunk 6 above)
Comment (gregogun) [reply sigle/src/types/index.ts:107]: would it be worth putting re-used types like `StatsData` in a separate types file within the stats module?

Comment (pradel) [reply sigle/src/types/index.ts:107]: Sure 👍

File: sigle/src/modules/analytics/stats/StatsFrame.tsx (hunk 3 above)
Comment (gregogun) [reply sigle/src/modules/analytics/stats/StatsFrame.tsx:147]: Great idea - updated and this is working much better (no longer a flash of no content between tabs), thanks!

**Review by pradel — APPROVED** [review]
Will merge on monday to avoid a Friday deployment :D

**Pull request merged**